src/components/ImageUpload/index.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+interface ImageUploadProps {
+  onImageSelect: (imageSrc: string) => void;
+}
+
+const ImageUpload: React.FC<ImageUploadProps> = ({ onImageSelect }) => {
+  const handleImageUpload = () => {
+    const input = document.createElement('input');
+    input.type = 'file';
+    input.accept = 'image/*';
+    input.onchange = (e) => {
+      const file = e.target.files && e.target.files[0];
+      if (file) {
+        const reader = new FileReader();
+        reader.onload = () => {
+          if (reader.result) {
+            onImageSelect(reader.result as string);
+          }
+        };
```

Comment:
```suggestion
reader.onload = (event: ProgressEvent<FileReader>) => {
          const result = event.target?.result;
          if (typeof result === 'string') {
            onImageSelect(result);
          }
        };
```

이런식으로 이벤트 타입을 명시적으로 사용해서 안정성을 높일 수 있습니다.
&
타입이 string일 경우에만 `onImageSelect`함수를 호출 할 수 있도록 해주세요.